ngraph/core/shape_inference/include/ctc_loss_shape_inference.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/op/ctc_loss.hpp>
+
+namespace ov {
+namespace op {
+namespace v4 {
+
+template <class T>
+inline void dynamic_inference(T& output_shape) {
+    OPENVINO_UNREACHABLE("This code should be executed only for PartialShape class");
+}
+
+template <>
+inline void dynamic_inference<ov::PartialShape>(ov::PartialShape& output_shape) {
+    output_shape = ov::PartialShape{Dimension::dynamic()};
```

Comment:
```suggestion
    output_shape = ov::PartialShape::dynamic();
```

Reply:
ov::PartialShape::dynamic() creates PartialShape with dynamic rank but here output_shape rank is 1 and Dimension is dynamic.